nixos/modules/services/home-automation/openthread-border-router.nix

```diff
@@ -0,0 +1,234 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.openthread-border-router;
+  logLevelMappings = {
+    "emerg" = 0;
+    "alert" = 1;
+    "crit" = 2;
+    "err" = 3;
+    "warning" = 4;
+    "notice" = 5;
+    "info" = 6;
+    "debug" = 7;
+  };
+  logLevel = lib.getAttr cfg.logLevel logLevelMappings;
+in
+{
+  meta.maintainers = with lib.maintainers; [ mrene ];
+
+  options.services.openthread-border-router = {
```

Comment:
I think we also need an `openFirewall` option:
```nix
networking.firewall.allowedUDPPorts = [
    49154
];
```

I don't think it makes sense to have firewall options for REST/web.

In a VM/lab environment, the admin should add `allowedTCPPorts` for `web.listenPort` / `rest.listenPort` themselves.

Reply:
Opened a discussion about listener ports upstream: https://github.com/orgs/openthread/discussions/11093